### PR TITLE
feat(BugFix Trace): Fix the span export problem

### PR DESCRIPTION
**Pull request by jccscopeo**
Fix the span export problem when exporting a bad span leads to a sqlalchemy connection stuck into a stale state:

- allow rollback when having span export errors
- improve logging of errors with span exports
- improve handling of errors/edge cases with spans + better logging
- Add integration tests for the trace

### CODE REVIEW

File: engine/trace/sql_exporter.py [hunk 1]
```diff
@@ -57,137 +59,324 @@ def convert_to_list(obj: Any) -> list[str] | None:
 
 class SQLSpanExporter(SpanExporter):
     def __init__(self):
-        self.session = get_session_trace()
+        # Don't create session here - create fresh session for each export
+        self.engine = None
+        self.Session = None
+        self._setup_engine()
```
Comment (dianascopeo) [inline engine/trace/sql_exporter.py:65]: why we need to create fresh session for each export? I dont get if this will solve the bug that we saw

File: engine/trace/sql_exporter.py [hunk 2]
```diff
@@ -61,133 +63,281 @@ def __init__(self):
...
         """Get org_id and org_llm_providers from ancestors of the span."""
-        while parent_id:
-            row = self.session.execute(
-                select(models.Span.attributes, models.Span.parent_id).where(models.Span.span_id == parent_id)
-            ).first()
-            if not row:
-                break
-            attributes, parent_id = row
-            if attributes:
+        try:
```
Comment (natacha-scopeo) [inline engine/trace/sql_exporter.py:66]: why 2 try nested?

Comment (jccscopeo) [reply engine/trace/sql_exporter.py]: you are right, it is useless. It was to catch some errors but it is useless in this function. I revert to the previous version
